### PR TITLE
introduced `number` object

### PR DESCRIPTION
**Pull request by Graur**
#742

### CODE REVIEW

**Comment by Graur** [comment]
@yegor256 It's the first part of the `number` object. Please, have a look

**Comment by Graur** [comment]
@yegor256 I've get a problem with 0 and 0.0 values, when I'm trying to figure out witch type they are belongs to.
We got both `TRUE`: 
```
0.eq 0
```
and
```
0.0.eq 0
```

Can I suggest that if we get `0`, it's always an `int` value? Because both their size is 8 bytes:
```
FLOAT BYTES [0, 0, 0, 0, 0, 0, 0, 0]
INT BYTES [0, 0, 0, 0, 0, 0, 0, 0]
```

**Comment by yegor256** [comment]
@Graur I think it's safer to assume that it's `float`, to make sure this works:

```
ceil.
  number 0.0
```

with an assumption that it's `int` this code will lead to error, right?

**Comment by Graur** [comment]
@yegor256 Thats the beauty of my solution to return `int` or `float` value instead of `bool`. You can do like this:
```
[] > check-type-positive-int
  assert-that > @
    (number 0).type
    .plus 4
    $.equal-to 4

```
and like this:
```
[] > check-type-positive-float
  assert-that > @
    (number 0.0).type
    .plus 4.0
    $.equal-to 4.0
```

Both are `TRUE`

**Comment by yegor256** [comment]
@Graur who will ever need this `type` attribute?)

**Comment by Graur** [comment]
@yegor256 I hope we will need it only inside `number` object.

**Comment by yegor256** [comment]
@Graur that's why I suggest calling it `is-int` or `is-float`, to make the code _inside_ `number` shorter and more readable

**Comment by Graur** [comment]
@yegor256 I think I can do as you suggested. I just need to add one `if` statement to check zero value to every object inside `number` and re-throw the decorated value `0` or `0.0`.

File: eo-runtime/src/test/eo/org/eolang/math/number-tests.eo [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+[] > check-is-int-negative-int
+  assert-that > @
+    (number -4623).is-int
+    $.equal-to TRUE
+    "check is-int of negative int"
+
+[] > check-is-int-with-zero
+  assert-that > @
+    (number 0).is-int
+    $.equal-to FALSE
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/math/number-tests.eo:45]: @yegor256 This is how it works. Is it ok?

File: eo-runtime/src/test/eo/org/eolang/math/number-tests.eo [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    $.equal-to 0
+
+[] > mod-into-while-and-seq-objects
+  [condition body] > while-loop
+    while. > @
+      condition
+      [i]
+        body > @
+
+  memory > n
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/math/number-tests.eo:298]: @Graur typo?

**Comment by yegor256** [comment]
@Graur looks much better now. I think you left some code by mistake there at the bottom of a file. Also, why didn't you move `signum` to `number` too?

**Comment by Graur** [comment]
@yegor256 I split it into two parts to improve readability of this PR. `mod`, `pow`, `as-float`,`as-string` and `signum` objects was left behind. But I can add it to this PR a little bit later.

File: eo-runtime/src/test/eo/org/eolang/math/number-tests.eo (hunk 2 above)
Comment (Graur) [reply eo-runtime/src/test/eo/org/eolang/math/number-tests.eo:298]: @yegor256 Indeed, thanks!

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**